core/misc/jquery.once.js

```diff
@@ -1,6 +1,9 @@
 /**
- * jQuery Once Plugin 1.2.6
- * http://github.com/robloach/jquery-once
+ * @file
+ * jQuery Once Plugin
+ *
+ * Forked from v1.2.6 of http://github.com/robloach/jquery-once, which was dual
+ * licensed.
  *
  * Dual licensed under the MIT and GPL licenses:
  *   http://www.opensource.org/licenses/mit-license.php
```

Comment:
The original part of the comment needs to be left as it is. The comment saying this file has been forked needs to be appended.

As it is written, it seems that the changes done to the original code are still dual licensed under MIT and GPL, which I am not sure is possible, in a repository that uses GPLv2 as license.

Reply:
> The original part of the comment needs to be left as it is. The comment saying this file has been forked needs to be appended.

That's the case now. The lines of the original license statement are untouched.

> As it is written, it seems that the changes done to the original code are still dual licensed under MIT and GPL, which I am not sure is possible, in a repository that uses GPLv2 as license.

That's exactly re reason, why I hesitated to keep the dual license part - our fork relies on the fact, that ONE of the licenses is GPL, so we can incorporate in our (GPL) code.

Can you please elaborate, what you expect me to do? Currently it reads  more like a catch-22. :laughing:

Reply:
The original comment starts with is the following lines.

```
 * jQuery Once Plugin 1.2.6
 * http://github.com/robloach/jquery-once
 *
 * Dual licensed under the MIT and GPL licenses:
 *   http://www.opensource.org/licenses/mit-license.php
```

Those still need to be the first lines in that comment. The comment saying the file has been forked needs to go after those lines; it could also say when the fork has been done, just in case the original file is later changed, even just to add a copyright line, or change the license used for the original file.

Keeping the lines separated allows to better understand the license for the existing code and the license for the changed/new lines. It is also what is normally done when files are forked from a different repository/source.

Reply:
Not sure, if I got it right this time. (Doesn't look right to me...)

However, another fix attempted. :slightly_smiling_face:

Reply:
It is how it was done on [system.tar.inc](https://github.com/backdrop/backdrop/blob/f32837d9a1ae3dcbc3bb0e27274bbd051ed89a95/core/modules/system/system.tar.inc#L4), where the original comment has been left, and a new comment has been added after that.

```php
/**
 * Backdrop-specific notes on this port from the original source.
 *
 * This file origin is Tar.php, release 1.4.9 (stable) with some code
 * from PEAR.php, release 1.10.10 (stable) both at http://pear.php.net.
 * To simplify future porting from pear of this file, you should not
 * do cosmetic or other non significant changes to this file.
 * The following changes have been done:
 *  Removed require_once 'PEAR.php'.
 *  Added definition of OS_WINDOWS taken from PEAR.php.
 *  Removed extends PEAR from class.
 *  Removed call parent:: __construct().
 *  Changed PEAR::loadExtension($extname) to this->loadExtension($extname).
 *  Added function loadExtension() taken from PEAR.php.
 *  Changed all calls of unlink() to backdrop_unlink().
 *  Changed $this->error_object = &$this->raiseError($p_message)
 *  to throw new Exception($p_message).
 */
```

That comment describes all the changes done to the original comment, but that is not necessary in this case, because the change is small.

Reply:
I think, the main difference here is, that Archive_Tar is still maintained (a bit), so possibly there might be a code change on their side to merge with the B version.

jQuery.once 1.x isn't maintained anymore. There won't be any update to merge in the future. And the code is simple, anyway. No comparison to the way more complex Archive_Tar.

However – are you satisfied with the way the comment is added?

Reply:
It's perfect!